python/cudf/cudf/_lib/scalar.pyx

```diff
@@ -92,6 +110,22 @@ cdef class Scalar:
                 f"Cannot convert value of type "
                 f"{type(value).__name__} to cudf scalar"
             )
+    
+    def get_device_value(self):
```

Comment:
Is this API needed anymore?

Reply:
This is used primarily inside the `value` property still, but only if the host side value isn't valid